core/src/main/scala/org/apache/spark/rdd/HadoopRDD.scala

```diff
@@ -196,6 +197,8 @@ class HadoopRDD[K, V](
     newInputFormat
   }
 
+  @transient private lazy val codecFactory = new CompressionCodecFactory(getJobConf())
```

Comment:
shall we create the codec factory inside `getPartitions`? According to the doc of `RDD#getPartitions`, this method will only be called once.